### PR TITLE
Use classes for input field generation

### PR DESCRIPTION
**Pull request by jechols**
- Adds interceptable / overridable pieces to the form field creation
- Uses more general "repeating-field" class for CSS to avoid relying on
  Ruby's class name
- Retains previous behavior for apps not customizing the form builders

### CODE REVIEW

**Comment by jcoyne** [comment]
:+1: Cool!

**Comment by jechols** [comment]
FYI, my third bullet point should be tested by somebody with a real app - the tests pass, and it seem to look the same in a relatively simple app, but I have to wonder if there is anything I might have missed in a more real-world situation.

**Comment by terrellt** [comment]
@jcoyne Can we merge this?

File: app/assets/javascripts/hydra-editor/editMetadata.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
 Blacklight.onLoad(function() {
-  $('.multi_value.form-group').manage_fields();
+  $('.repeating-field.form-group').manage_fields();
```
Comment (awead) [inline app/assets/javascripts/hydra-editor/editMetadata.js]: @jechols why the change in css class? This would break current implementations, I believe.

Comment (jechols) [inline app/assets/javascripts/hydra-editor/editMetadata.js]: In the wrapper html, I actually add a new class.  `.multi_value` is simple_form magic based on the input type's name, and it'll be done the same as it already is today.  The trick is that if you define a new input type, but want the same logic (which we're very likely to need), the custom input type has to lie to simple form about its type name in order to get that class in the CSS

So if there's code in the wild that relies on .multi_value for custom styling, it'll be fine - that class is still there since it's simple_form magic.

**Comment by jechols** [comment]
Closing to revisit this (I have concerns about how I did this thing last month relative to what the community might need or want)
